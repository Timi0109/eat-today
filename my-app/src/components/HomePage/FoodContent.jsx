import React,{useEffect, useState} from "react";
import styles from "./FoodContent.module.css";
import { ArrowRight } from 'react-bootstrap-icons';

const FoodContent = () => {
    const [content, setContent] = useState([])
    const [fetchedData, setFetchedData] = useState([]);
    
    useEffect( () => {
        const fetchData  = async() => {
            const response = await fetch("https://run.mocky.io/v3/e2608e4d-c9e8-4cd6-8ccd-21a0aeedb3d5");
            const responseJson = await response.json();
            setFetchedData(responseJson.food);
        };
        if (!fetchedData.length) {
            fetchData();
        }
    },[fetchedData]);

    const pickFood = () => {
      const num = Math.floor(Math.random() * fetchedData.length);
      setContent(fetchedData[num]);
    }

    return (
        <div className={styles.foodContainer}>
            <div className={styles.randContainer}>
                <button className={styles.pickFoodBtn} onClick={() => pickFood()}>Pick Food</button>
                <p className={styles.tips}>No idea what eat today? Let's pick randomly!</p>
            </div>
                <ArrowRight className={styles.arrowIcon} size={96} />
            <div className={styles.choosedFoodContainer}>
               {content.name ?<p className={styles.foodName}>{content.name}</p> :null}
               {content.name ?<img className={styles.pickedFood} src={content.url} alt="picked-img"></img> 
                            //  :null
                             :<img className={styles.pickedFoodGif} src="https://i.pinimg.com/originals/f0/ca/90/f0ca90dd6924e009d86f4421cf2032b5.gif" alt="header png"></img>
                             }
            </div>
        </div>
    )
}


export default FoodContent