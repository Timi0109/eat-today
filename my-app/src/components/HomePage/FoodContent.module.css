.foodContainer{
    text-align: center;
    margin-top: 5rem;
    width: auto;
    height: auto;
    display: grid;
    margin-left: 15%;
    grid-template-columns: 25% 25% 40% 10%; 
}

.randContainer{
    text-align: center;
    margin-top: 15rem;
    display: grid;
    grid-template-rows: 80px;
    grid-template-columns: 100%;
    
}

.arrowIcon{
    margin-top: 15rem;
    margin-left: auto;
    margin-right: auto; 
    color: #f06292;
}

.choosedFoodContainer{
    border: 2px dotted pink;
    font-size: 24px;
    margin-top: 8rem;
}

.foodName{
    font-size: 30px;
    color: gray;
    font-weight: bold;
}

.pickedFood{
    width: 100%;
    height: auto;
}

.pickedFoodGif{
    width: 100%;
    height: auto;
    
}

.pickFoodBtn {
    color: #f06292;
    width: 200px;
    height: 49px;
    border-radius: 1em;
    font-size: 14px;
    background-color: transparent;
    font-weight: bold;
    border-color: #f3bad6;
    outline: none;
    margin-top: 20px;
}
  
.pickFoodBtn:hover{
    background-color: #f06292;
    color: white;
    outline: none;
}

.tips{
    margin-top: 30px;
    color: #f06292;
    font-weight: bold;
    
}

.foodSet{
  height: 100%;
  float: left;
  opacity: 0.6;
  width: 100%;
}

@media (max-width: 980px){
    .arrowIcon{
        display: none;
    }
    .foodContainer{
        display: flex;
        flex-direction: column;
    }
  }