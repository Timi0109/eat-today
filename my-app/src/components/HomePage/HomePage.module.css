.homePage{
  background-image: url("https://cdn.dribbble.com/users/1012997/screenshots/14086465/hunger-station-dribbble_4x.png");
  background-repeat: no-repeat;
  background-position: top;
}

.header{
  height: 100px;
  width: 100%;
}

.container{
  height: 700px;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

.pageTitleContainer{
  margin-top: 100px;
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  height: 400px;
  position: relative;
}

.pagetitle{
  position: absolute;
  color: #e05297;
  text-align: center;
  font-family: 'Cabin', sans-serif;
  font-size: 50px;
  margin-left: auto;
  margin-right: auto;
  width: 30%;
  bottom: 80px;
  left: 28%;

}

.pagetitleTwo{
  position: absolute;
  color: #e05297;
  text-align: center;
  font-family: 'Cabin', sans-serif;
  font-size: 50px;
  margin-left: auto;
  margin-right: auto;
  width: 30%;
  bottom: 10px;
  left: 37%;
}




@media (max-width: 400px){
  .pagetitle{
    visibility: hidden;
  }
  .pagetitleTwo{
    visibility: hidden;
  }
}

@media (max-width: 1400px){
  .pagetitle{
    visibility: hidden;
  }
  .pagetitleTwo{
    visibility: hidden;
  }
}