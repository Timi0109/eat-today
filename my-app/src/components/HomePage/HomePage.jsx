import React from "react"
import MenuBar from "../MenuBar/MenuBar.jsx";
import styles from "./HomePage.module.css";
import FoodContent from "./FoodContent";
import Title from "../Title.jsx";
import Footer from "../Footer/Footer.jsx";
import {userHistory, useLocation, useHistory} from "react-router-dom";

const HomePage = () => {
    let location = useLocation();
    let history = useHistory();
    console.log(location);
    console.log(history);
    return (
        <div className={styles.homePage}>
            <header className={styles.header}>
                <MenuBar></MenuBar> 
                <Title></Title>
            </header>
            <div className={styles.pageTitleContainer}>
                <p className={styles.pagetitle}>Welcome To,</p>
                <p className={styles.pagetitleTwo}>Eat Today</p>
            </div>
            <div className={styles.container}>
                <FoodContent></FoodContent>
            </div>
            <Footer></Footer>
        </div>
        
    )
}


export default HomePage 