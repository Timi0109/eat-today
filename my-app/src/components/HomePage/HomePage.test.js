import React from "react";
import HomePage from "./HomePage";

describe("HomePage tests", () => {
  it("renders correctly", () => {
    const { container } = render(<HomePage></HomePage>);
    expect(container).toMatchSnapshot();
  });

});
