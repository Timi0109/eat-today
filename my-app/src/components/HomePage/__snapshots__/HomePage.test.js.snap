// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests renders correctly 1`] = `
<div>
  <div
    class="homePage"
  >
    <header
      class="header"
    >
      <nav
        class="Nav"
      >
        <a
          class="Nav-child"
          href="/"
        >
          Home
        </a>
        <a
          class="Nav-child"
          href="/recipes"
        >
          Recipe
        </a>
        <a
          class="Nav-child"
          href="/login"
        >
          LogIn
        </a>
      </nav>
      <h2
        class="title"
      >
        EatToday
      </h2>
    </header>
    <div
      class="pageTitleContainer"
    >
      <p
        class="pagetitle"
      >
        Welcome To,
      </p>
      <p
        class="pagetitleTwo"
      >
        Eat Today
      </p>
    </div>
    <div
      class="container"
    >
      <div
        class="foodContainer"
      >
        <div
          class="randContainer"
        >
          <button
            class="pickFoodBtn"
          >
            Pick Food
          </button>
          <p
            class="tips"
          >
            No idea what eat today? Let's pick randomly!
          </p>
        </div>
        <svg
          class="arrowIcon"
          fill="currentColor"
          height="96"
          viewBox="0 0 16 16"
          width="96"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="choosedFoodContainer"
        >
          <img
            alt="header png"
            class="pickedFoodGif"
            src="https://i.pinimg.com/originals/f0/ca/90/f0ca90dd6924e009d86f4421cf2032b5.gif"
          />
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <p
        class="title"
      >
        EATTODAY.COM
      </p>
      <div
        class="container"
      >
        <div>
          <p
            class="subTitle"
          >
            About Us
          </p>
          <p>
            Home
          </p>
          <p>
            How It Works
          </p>
        </div>
        <div>
          <p
            class="subTitle"
          >
            Contact Us
          </p>
          <p>
            Linkedin
          </p>
          <p>
            Twitter
          </p>
          <p>
            Wechat
          </p>
        </div>
      </div>
      <p
        class="right"
      >
        @2020 EATTODAY.
      </p>
    </div>
  </div>
</div>
`;
